### PR TITLE
change field versioning passes

### PR DESCRIPTION
**Pull request by twicki**
## Technical Description

Rework the testing framework. Changes the field_versioning test to only use IIR and not go through SIR.

### Resolves / Enhances

Work on https://github.com/MeteoSwiss-APN/dawn/issues/702

### CODE REVIEW

**Comment by jdahm** [comment]
I like the consistent naming of the IIRs for testing passes like this. Good work!

**Comment by twicki** [comment]
launch jenkins

**Comment by twicki** [comment]
launch jenkins

**Comment by twicki** [comment]
launch jenkins

File: dawn/test/unit-test/dawn/Optimizer/Passes/TestPassFieldVersioning.cpp [hunk 1]
```diff
@@ -11,71 +11,217 @@
...
   dawn::OptimizerContext::OptimizerContextOptions options_;
+  DiagnosticsEngine diagnostics_;
   std::unique_ptr<OptimizerContext> context_;
 
   void raceConditionTest(const std::string& filename) {
-    CompilerUtil::clearDiags();
+    std::string fullFileName = filename;
+    if(!TestEnvironment::path_.empty() && filename.at(0) != '/') {
+      fullFileName = TestEnvironment::path_ + "/" + filename;
+    }
```
[the remark this answers is not in the document]
Comment (twicki) [reply dawn/test/unit-test/dawn/Optimizer/Passes/TestPassFieldVersioning.cpp]: In my understanding `fs::path` is used to extend files with their absolute path whereas here we read in the argument passed to the test and append the filename with that. Are these not different things?

[the remark this answers is not in the document]
Comment (Stagno) [reply dawn/test/unit-test/dawn/Optimizer/Passes/TestPassFieldVersioning.cpp]: I was thinking about using it to do the check that filename is absolute and to safely append the filename (it shouldn't matter if `path_` ends with "/" or not):
```
fs::path filename = ...;
 fs::path fullFileName = filename;
    if(!TestEnvironment::path_.empty() && filename.is_relative()) {
      fullFileName = TestEnvironment::path_ / filename;
    }
```
(assuming `TestEnvironment::path_` is also a `fs::path`)

**Comment by twicki** [comment]
launch jenkins

**Pull request merged**